### DISCUSSION TITLE
How to create  branching activities in elsa 3.0

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5592] [Q&A]
This is my input model
public class InputModel
{
  public string Property1{get;set;}
  public bool Property2{get;set;}
}
this is input model.
Workflow is below way

public class CustomWorkflow : IWorkflow
{
public void Build(IWorkflowBuilder builder)
{
builder
.StartWith()
         .Then<Activity1>()
         .Then<Activity2>();
         .Then<Activity3>()
         .Then<Activity4>();
}
}

Activity 2 should only executed when  Property2 value is true. Please help me in resolving this issue

**Comment by glime-ai** [comment]
To conditionally execute `Activity2` based on the value of `Property2` from your `InputModel`, you can use an `If` activity or a similar conditional construct provided by Elsa. Here's how you can modify your workflow to include this logic:

```csharp
public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
            .If(context => context.GetInput<InputModel>().Property2)
                .Then<Activity2>() // This will only execute if Property2 is true
            .Else()
                .Then<Activity3>() // You can continue with Activity3 if Property2 is false
            .Then<Activity4>();
    }
}
```

In this example, the `If` activity evaluates the `Property2` property of the `InputModel` object. If `Property2` is `true`, `Activity2` is executed. Otherwise, the workflow continues with `Activity3`. After `Activity2` or `Activity3`, the workflow proceeds to `Activity4`.

Please note that the actual syntax for accessing the input and the conditional logic may vary depending on the version of Elsa you are using and how you have set up your input model within the workflow context. The above code assumes that you have a way to retrieve the `InputModel` from the workflow context.

If you need further assistance, please provide more details about how you are passing the `InputModel` to the workflow and which version of Elsa you are using.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
